.github/workflows/doc-tests.yml

```diff
@@ -11,20 +11,48 @@ permissions:
 
 jobs:
   doc-test:
-    name: Sphinx-${{ matrix.builder }}
+    name: Sphinx Documentation Tests
     runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental || false }}
-    strategy:
-      fail-fast: false
-      matrix:
-        builder: [ html, linkcheck ]
-        include:
-          # Run default html builder with warnings as error
-          - builder: html
-            args: -W
+
     steps:
       - uses: actions/checkout@v4
-      - name: Run sphinx builder ${{ matrix.builder }}
-        run: |
-          pip install hatch
-          hatch run docs:${{ matrix.builder }} ${{ matrix.args }}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+      - name: Set PIPX_HOME
+        run: echo "PIPX_HOME=${{ github.workspace }}/.pipx" >> $GITHUB_ENV
+
+      - name: Cache pipx environments
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/.pipx
+          # pipx is creating a cachedir.tag with unique
+          key: ${{ runner.os }}-pipx-home-${{ hashFiles('.pipx/.cache/CACHEDIR.TAG') }}
+          restore-keys: |
+            ${{ runner.os }}-pipx-home-
```

Comment:
Please use `v4` . Also we can get rid of `runner.os` part.

For this one, the `.pipx` will not work because we do not track such a file. Instead why not keep the default `pip` cache for this part? Afaiu `pipx` would only help if we want to minimize the run to 1 line, but we are not doing that here.

Reply:
we're caching the full hatch environment. I don't think we can do that with just pip, but happy to be taught otherwise.

Reply:
ok, scrapping this bad idea :)